src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -1253,3 +1253,104 @@ def get_parent_proc_name():
         parent_proc_name = _get_parent_proc_name()
         setattr(get_parent_proc_name, "return_value", parent_proc_name)
     return getattr(get_parent_proc_name, "return_value")
+
+
+def log_cmd_history(command, args):
+    import os
+    from knack.util import ensure_dir
+    from azure.cli.core.extension import get_extension, ExtensionNotInstalledException
+    from azure.cli.core._environment import get_config_dir
+
+    if command == 'next':
+        return
+
+    # Determine whether "az next" has been installed.
+    # At present, command execution log is only recorded when "az next" is installed
+    try:
+        az_next_is_installed = get_extension("next")
+    except ExtensionNotInstalledException:
+        az_next_is_installed = False
+
+    if not az_next_is_installed:
+        return
+
+    # For commands that do not require logging, we can pass in the parameter "--no-log" to control
+    if '--no-log' in args:
```

Comment:
As I know, there are settings collect_telemetry and enable_log_file in our config. Will these setting also influence if you will start logging?

Reply:
At present, this method is only available for `az next` to use, so it has a independent control logic in the scenario of `az next`, which will not be affected by these settings.
I'll migrate this method to extension in an event driven way in the next sprint

Reply:
I see. Please also consider if user doesn't enable logging in config. In such scenario, we may not also log for `az next` to avoid policy violations.
@fengzhou-msft do you have any idea for logging to file setting?